lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -54,67 +55,104 @@ export class Vulnerabilities {
     config: RenovateConfig,
     packageFiles: Record<string, PackageFile[]>
   ): Promise<void> {
+    const dependencyVulnerabilities = await this.fetchDependencyVulnerabilities(
+      config,
+      packageFiles
+    );
+
+    config.packageRules ??= [];
+    for (const {
+      vulnerabilities,
+      versioningApi,
+    } of dependencyVulnerabilities) {
+      const groupPackageRules: PackageRule[] = [];
+      for (const vulnerability of vulnerabilities) {
+        const rule = this.vulnerabilityToPackageRules(vulnerability);
+        if (is.nullOrUndefined(rule)) {
+          continue;
+        }
+        groupPackageRules.push(rule);
+      }
+      this.sortByFixedVersion(groupPackageRules, versioningApi);
+
+      config.packageRules.push(...groupPackageRules);
+    }
+  }
+
+  async fetchVulnerabilities(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>
+  ): Promise<Vulnerability[]> {
+    const groups = await this.fetchDependencyVulnerabilities(
+      config,
+      packageFiles
+    );
+    return groups.flatMap((group) => group.vulnerabilities);
+  }
+
+  async fetchDependencyVulnerabilities(
```

Comment:
```suggestion
  private async fetchDependencyVulnerabilities(
```

can this be private ?